protocol/x/vault/keeper/orders.go

```diff
@@ -30,10 +32,82 @@ const (
 )
 
 // RefreshAllVaultOrders refreshes all orders for all vaults by
-// TODO(TRA-134)
 // 1. Cancelling all existing orders.
 // 2. Placing new orders.
 func (k Keeper) RefreshAllVaultOrders(ctx sdk.Context) {
+	// Iterate through all vaults.
+	totalSharesIterator := k.getTotalSharesIterator(ctx)
+	defer totalSharesIterator.Close()
+	for ; totalSharesIterator.Valid(); totalSharesIterator.Next() {
+		var vaultId types.VaultId
+		k.cdc.MustUnmarshal(totalSharesIterator.Key(), &vaultId)
+		var totalShares types.NumShares
+		k.cdc.MustUnmarshal(totalSharesIterator.Value(), &totalShares)
+
+		// Skip if TotalShares is non-positive.
+		if totalShares.NumShares.Cmp(dtypes.NewInt(0)) <= 0 {
+			continue
+		}
+
+		// Refresh orders depending on vault type.
+		// Currently only supported vault type is CLOB.
+		switch vaultId.Type {
+		case types.VaultType_VAULT_TYPE_CLOB:
+			err := k.RefreshVaultClobOrders(ctx, vaultId)
+			if err != nil {
+				log.ErrorLogWithError(ctx, "Failed to refresh vault clob orders", err, "vaultId", vaultId)
+			}
+		default:
+			log.ErrorLog(ctx, "Failed to refresh vault orders: unknown vault type", "vaultId", vaultId)
+		}
+	}
+}
+
+// RefreshVaultClobOrders refreshes orders of a CLOB vault.
+func (k Keeper) RefreshVaultClobOrders(ctx sdk.Context, vaultId types.VaultId) (err error) {
+	// Cancel CLOB orders from last block.
+	ordersToCancel, err := k.GetVaultClobOrders(
+		ctx.WithBlockHeight(ctx.BlockHeight()-1),
+		vaultId,
+	)
+	if err != nil {
+		log.ErrorLogWithError(ctx, "Failed to get vault clob orders to cancel", err, "vaultId", vaultId)
+		return err
+	}
+	for _, order := range ordersToCancel {
+		if _, exists := k.clobKeeper.GetLongTermOrderPlacement(ctx, order.OrderId); exists {
```

Comment:
Is this method able to find orders that are partially filled?
For example, if the vault had half of its order filled, it would still get canceled in RefreshVaultClobOrders, correct?